lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,249 @@
+
+/// This structure contains air dragging properties.
+#[derive(Default, Clone, Copy)]
+pub struct DragProperties {
+    amount : f32
+}
+
+impl DragProperties {
+    pub fn new(amount:f32) -> Self {
+        Self { amount }
+    }
+}
+
+// === Getters ===
+
+impl DragProperties {
+    pub fn amount(&self) -> f32 {
+        self.amount
+    }
+}
+
+// === Setters ===
+
+impl DragProperties {
+    pub fn set_amount(&mut self, amount:f32) {
+        self.amount = amount
+    }
+}
+
+// ========================
+// === SpringProperties ===
+// ========================
+
+/// This structure contains spring physics properties.
+#[derive(Debug, Clone, Copy)]
+pub struct SpringProperties {
+    coefficient : f32,
+    fixed_point : Vector3<f32>
+}
+
+impl Default for SpringProperties {
+    fn default() -> Self {
+        Self::new(zero(),zero())
+    }
+}
+
+impl SpringProperties {
+    pub fn new(coefficient:f32, fixed_point:Vector3<f32>) -> Self {
+        Self { coefficient,fixed_point }
+    }
+}
+
+// === Getters ===
+
+impl SpringProperties {
+    pub fn coefficient(&self) -> f32          { self.coefficient }
+    pub fn fixed_point(&self) -> Vector3<f32> { self.fixed_point }
+}
+
+// === Setters ===
+
+impl SpringProperties {
+    pub fn set_coefficient(&mut self, coefficient:f32)          { self.coefficient = coefficient }
+    pub fn set_fixed_point(&mut self, fixed_point:Vector3<f32>) { self.fixed_point = fixed_point }
+}
+
+// ============================
+// === KinematicProperties ===
+// ============================
+
+/// This structure contains kinematics properties.
+#[derive(Debug, Clone, Copy)]
+pub struct KinematicsProperties {
+    position     : Vector3<f32>,
+    velocity     : Vector3<f32>,
+    acceleration : Vector3<f32>,
+    mass         : f32
+}
+
+impl Default for KinematicsProperties {
+    fn default() -> Self {
+        Self::new(zero(),zero(),zero(),zero())
+    }
+}
+
+impl KinematicsProperties {
+    pub fn new
+    (position:Vector3<f32>, velocity:Vector3<f32>, acceleration:Vector3<f32>, mass:f32) -> Self {
+        Self { position,velocity,acceleration,mass }
+    }
+}
+
+// === Getters ===
+
+impl KinematicsProperties {
+    pub fn velocity    (&self) -> Vector3<f32> { self.velocity }
+    pub fn acceleration(&self) -> Vector3<f32> { self.acceleration }
+    pub fn mass        (&self) -> f32          { self.mass }
+}
+
+// === Setters ===
+
+impl KinematicsProperties {
+    pub fn set_velocity(&mut self, velocity:Vector3<f32>) {
+        self.velocity = velocity
+    }
+
+    pub fn set_acceleration(&mut self, acceleration:Vector3<f32>) {
+        self.acceleration = acceleration
+    }
+
+    pub fn set_mass(&mut self, mass:f32) {
+        self.mass = mass
+    }
+}
+
+impl HasPosition for KinematicsProperties {
+    fn position    (&self) -> Vector3<f32>            { self.position }
+    fn set_position(&mut self, position:Vector3<f32>) { self.position = position }
+}
+
+// =============================
```

Comment:
This is very very minor, but please remember about it - before these big headers we put 3 empty lines

Reply:
https://github.com/luna/basegl/blob/master/docs/style-guide.md#vertical-spacing defines 2 empty lines between sections.

Reply:
Thanks for cathcing it. I need to fix the style guide :(